src/test/java/pl/zaprogramujzycie/mma/controllers/MedicineControllerTest.java

```diff
@@ -0,0 +1,250 @@
+package pl.zaprogramujzycie.mma.controllers;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import net.minidev.json.JSONArray;
+import org.flywaydb.core.Flyway;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.*;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import pl.zaprogramujzycie.mma.dto.request.MedicineRequest;
+import pl.zaprogramujzycie.mma.dto.response.MedicineResponse;
+import pl.zaprogramujzycie.mma.dto.response.MedicinesResponse;
+
+import java.net.URI;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class MedicineControllerTest {
+
+    @Autowired
+    TestRestTemplate restTemplate;
+
+
+    private MedicineResponse medicineDTO;
+
+    private List<MedicineResponse> medicineDTOs = new ArrayList<>();
+
+
+    @BeforeEach
+    public void setup()
+    {
+        var medDTO1 = new MedicineResponse(100L, "Med1", OffsetDateTime.parse("2023-06-15T21:27:17.289601+02"), 6, "100");
+        var medDTO2 = new MedicineResponse(101L, "Med2", OffsetDateTime.parse("2023-06-15T21:27:17.289601+02"), 6, "100");
+        var medDTO3 = new MedicineResponse(102L, "Med3", OffsetDateTime.parse("2023-06-15T21:27:17.289601+02"), 6, "100");
+        this.medicineDTO = medDTO1;
+        medicineDTOs.add(medDTO1);
+        medicineDTOs.add(medDTO2);
+        medicineDTOs.add(medDTO3);
+    }
+
+    @BeforeEach
+    void clearDatabase(@Autowired Flyway flyway) {
+        flyway.clean();
+        flyway.migrate();
+    }
+
+    @Test
+    void shouldReturnAMedicineWhenDataIsSaved() {
```

Comment:
//give//when//then? poczytaj sobie o testach cos

Reply:
Znam. Jest to konwencja nie wymóg.